app/src/main/java/com/example/sharewhatyoucanproject/LoginActivity.kt

```diff
@@ -62,170 +56,155 @@ class LoginActivity : AppCompatActivity() {
+                                            .build()
+                                    user?.updateProfile(profileUpdates)
+                                        ?.addOnCompleteListener(
+                                            OnCompleteListener { task3 ->
+                                                if (task3.isSuccessful) {
+                                                    pd.dismiss()
+                                                    if (type == 1) {
+                                                        myEdit.putInt("selector", 1)
+                                                        myEdit.commit()
+                                                        val i = Intent(
+                                                            this@LoginActivity,
+                                                            SelectUserActivity::class.java
+                                                        )
+                                                        startActivity(i)
+                                                        finishAffinity()
+                                                    } else if (type == 2) {
+                                                        myEdit.putInt("selector", 2)
+                                                        myEdit.commit()
+                                                        val i = Intent(
+                                                            this,
+                                                            DrawerActivity2::class.java
+                                                        )
+                                                        startActivity(i)
+                                                        finishAffinity()
+                                                    }
+                                                } else {
+                                                    pd.dismiss()
+                                                    Toast.makeText(
+                                                        applicationContext,
+                                                        "Failed " + task3.exception,
+                                                        Toast.LENGTH_SHORT
+                                                    ).show()
+                                                }
+                                            }
+                                        )
+                                } else {
+                                    pd.dismiss()
+                                    Toast.makeText(
+                                        applicationContext,
+                                        "Failed " + task2.exception,
+                                        Toast.LENGTH_SHORT
+                                    )
+                                        .show()
+                                }
+                            }
```

Comment:
This is a lot.  Can you push this into a helper function?   Otherwise (a) hard to read and understand and (b) hard to unit test.